components/Toast/toast.tsx

```diff
@@ -0,0 +1,119 @@
+"use client"
+
+import * as ToastPrimitives from "@radix-ui/react-toast"
+import { cva, type VariantProps } from "class-variance-authority"
+import { RxCross1 } from "react-icons/rx";
+import * as React from "react"
+
+const ToastProvider = ToastPrimitives.Provider
+
+const ToastViewport = React.forwardRef<
+	React.ElementRef<typeof ToastPrimitives.Viewport>,
+	React.ComponentPropsWithoutRef<typeof ToastPrimitives.Viewport>
+>(({ className, ...props }, ref) => (
+	<ToastPrimitives.Viewport
+		ref={ref}
+		className={`fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px] ${className}`}
+		{...props}
+	/>
+))
+ToastViewport.displayName = ToastPrimitives.Viewport.displayName
+
+const toastVariants = cva(
```

Comment:
It is a new library that I have added. CVA stands for "class-variance-authority". It just helps in implementing "variants" for components by safely handling the classNames associated with them. 

The primary purpose of including this in the project (after trying to avoid it for so long) is because it makes it easier to directly copy the "manual" implementation provided by Shadcn. I have still not included the `cn` library, but I might do that as well in the future so that we can directly copy-paste from shadcn.